src/RadhydroPulseGrey/test_radhydro_pulse_grey.cpp

```diff
@@ -0,0 +1,425 @@
+	static constexpr bool is_radiation_enabled = true;
+	// face-centred
+	static constexpr bool is_mhd_enabled = false;
+	static constexpr int nGroups = 1;
+};
+
+AMREX_GPU_HOST_DEVICE
+auto compute_initial_Tgas(const double x) -> double
+{
+	// compute temperature profile for Gaussian radiation pulse
+	const double sigma = width;
+	return T0 + (T1 - T0) * std::exp(-x * x / (2.0 * sigma * sigma));
+}
+
+AMREX_GPU_HOST_DEVICE
+auto compute_exact_rho(const double x) -> double
+{
+	// compute density profile for Gaussian radiation pulse
+	auto T = compute_initial_Tgas(x);
+	return rho0 * T0 / T + (a_rad * mu / 3. / k_B) * (std::pow(T0, 4) / T - std::pow(T, 3));
+}
+
+template <> AMREX_GPU_HOST_DEVICE auto RadSystem<PulseProblem>::ComputePlanckOpacity(const double rho, const double Tgas) -> quokka::valarray<double, nGroups_>
+{
+	const double sigma = 3063.96 * std::pow(Tgas / T0, -3.5);
+	quokka::valarray<double, nGroups_> kappaPVec{};
+	kappaPVec.fillin(sigma / rho);
+	return kappaPVec;
+}
+template <>
+AMREX_GPU_HOST_DEVICE auto RadSystem<AdvPulseProblem>::ComputePlanckOpacity(const double rho, const double Tgas) -> quokka::valarray<double, nGroups_>
+{
+	const double sigma = 3063.96 * std::pow(Tgas / T0, -3.5);
+	quokka::valarray<double, nGroups_> kappaPVec{};
+	kappaPVec.fillin(sigma / rho);
+	return kappaPVec;
+}
+
+template <>
+AMREX_GPU_HOST_DEVICE auto RadSystem<PulseProblem>::ComputeFluxMeanOpacity(const double rho, const double Tgas) -> quokka::valarray<double, nGroups_>
+{
+	const double sigma = 101.248 * std::pow(Tgas / T0, -3.5);
+	quokka::valarray<double, nGroups_> kappaPVec{};
+	kappaPVec.fillin(sigma / rho);
+	return kappaPVec;
+}
+template <>
+AMREX_GPU_HOST_DEVICE auto RadSystem<AdvPulseProblem>::ComputeFluxMeanOpacity(const double rho, const double Tgas) -> quokka::valarray<double, nGroups_>
+{
+	return RadSystem<PulseProblem>::ComputeFluxMeanOpacity(rho, Tgas);
+}
+
+template <> void RadhydroSimulation<PulseProblem>::setInitialConditionsOnGrid(quokka::grid grid_elem)
+{
+	// extract variables required from the geom object
+	amrex::GpuArray<amrex::Real, AMREX_SPACEDIM> const dx = grid_elem.dx_;
+	amrex::GpuArray<amrex::Real, AMREX_SPACEDIM> prob_lo = grid_elem.prob_lo_;
+	amrex::GpuArray<amrex::Real, AMREX_SPACEDIM> prob_hi = grid_elem.prob_hi_;
+	const amrex::Box &indexRange = grid_elem.indexRange_;
+	const amrex::Array4<double> &state_cc = grid_elem.array_;
+
+	amrex::Real const x0 = prob_lo[0] + 0.5 * (prob_hi[0] - prob_lo[0]);
+
+	// loop over the grid and set the initial condition
+	amrex::ParallelFor(indexRange, [=] AMREX_GPU_DEVICE(int i, int j, int k) {
+		amrex::Real const x = prob_lo[0] + (i + static_cast<amrex::Real>(0.5)) * dx[0];
+		const double Trad = compute_initial_Tgas(x - x0);
+		const double Erad = a_rad * std::pow(Trad, 4);
+		const double rho = compute_exact_rho(x - x0);
+		const double Egas = quokka::EOS<PulseProblem>::ComputeEintFromTgas(rho, Trad);
+		const double v0 = v0_nonadv;
```

Comment:
Is there a reason to write it this way instead of just setting all the elements to the values appropriate for v0 = 0? I ask just because when I read over this source code I was initially very confused why the initial condition setup routines for advecting and non-advecting were identical, and my confusion only resolved when I noticed that the two routines differed by exactly two characters.

Reply:
Yes, I can do that. v0_nonadv is use only once and I can write v0 = 0 explicitly.